api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -478,8 +478,8 @@ private String getDuplicatePatientsSQLString(List<String> attributes) {
 				innerFields.add("person1." + attribute);
 			} else if (personNameFieldNames.contains(attribute)) {
 				if (!outerSelect.toString().contains("person_name")) {
-					outerSelect.append("inner join person_name t3 on t2.person_id = t3.person_id ");
-					innerSelect.append("inner join person_name pn1 on person1.person_id = pn1.person_id ");
+					outerSelect.append("inner join person_name t3 on t1.patient_id = t3.person_id ");
```

Comment:
Do you mind explaining this change?

Reply:
Sure. Before that's how the SQL string looked like:
`select distinct t1.patient_id from patient t1 inner join person_name t3 on t2.person_id = t3.person_id , (Select pn1.family_name from patient p1 inner join person_name pn1 on person1.person_id = pn1.person_id  where p1.voided = false  group by pn1.family_name having count(*) > 1 order by pn1.family_name) t5 where  t3.family_name = t5.family_name;`

It had two syntax errors:
- first, there is no t2 alias defined anywhere,  I've changed that to t1 which is an alias for patient table. It also doesn't contain person_id column - now it's called patient_id
- second, there is no person2 alias defined, I've changed that to p1 which is an alias used in this inner join. It also doesn't contain person_id column - now it's called patient_id

That's how the SQL string looks now:
`select distinct t1.patient_id from patient t1 inner join person_name t3 on t1.patient_id = t3.person_id , (Select pn1.family_name from patient p1 inner join person_name pn1 on p1.patient_id = pn1.person_id  where p1.voided = false  group by pn1.family_name having count(*) > 1 order by pn1.family_name) t5 where  t3.family_name = t5.family_name;`

Basically, the initial thought behind this was to inner join `person_name` table with `patient` table. The previous code used wrong aliases and column names so I fixed those.